src/Catel.MVVM/Catel.MVVM.Shared/Collections/Extensions/SuspensionContextExtensions.cs

```diff
@@ -46,48 +46,48 @@ public static ICollection<NotifyRangedCollectionChangedEventArgs> CreateEvents<T
         }
 
         /// <summary>
-        /// Creates the adding event args list.
+        /// Creates the none event args list.
         /// </summary>
         /// <param name="suspensionContext">The suspension context.</param>
         /// <typeparam name="T">The type of collection item.</typeparam>
         /// <returns>The <see cref="ICollection{NotifyRangedCollectionChangedEventArgs}"/>.</returns>
-        public static ICollection<NotifyRangedCollectionChangedEventArgs> CreateAddingEvents<T>(this SuspensionContext<T> suspensionContext)
+        public static ICollection<NotifyRangedCollectionChangedEventArgs> CreateNoneEvents<T>(this SuspensionContext<T> suspensionContext)
         {
-            Argument.IsNotNull(nameof(suspensionContext), suspensionContext);
-            Argument.IsValid(nameof(suspensionContext.Mode), suspensionContext.Mode, mode => mode == SuspensionMode.Adding);
+            Argument.IsValid(nameof(suspensionContext), suspensionContext, context => context == null || context.Mode == SuspensionMode.None);
 
-            return new List<NotifyRangedCollectionChangedEventArgs> { new NotifyRangedCollectionChangedEventArgs(suspensionContext.ChangedItems, suspensionContext.ChangedItemIndices, suspensionContext.Mode) };
+            return new List<NotifyRangedCollectionChangedEventArgs> { new NotifyRangedCollectionChangedEventArgs() };
```

Comment:
This might be heavy on the GC if this is being used a lot. Could this be replaced by ArrayShim.Empty<NotifyRangedCollectionChangedEventArgs>(); or do we really need to instance? If so, we could also choose to cache it.

Reply:
This is old code. I have reorded the functions to another order to match the order in the enum.
And so the "funny" changes appear here ... I have changed nothing.

Netherless, we need a list with one element. How could this be replaced with an empty array?

Reply:
We could cache the instance with a single item, but let's leave as-is. We have memory optimization covered in another ticket.

Reply:
And than always call clear befor reusing? That was the only thing I could image.